src/transport/SecureSessionMgr.h

```diff
@@ -68,6 +68,8 @@ class EncryptedPacketBufferHandle final : private System::PacketBufferHandle
 
     uint32_t GetMsgId() const;
 
+    PacketBufferHandle Retain() const { return PacketBufferHandle::Retain(); }
```

Comment:
This is violating the main invariant of this class, which is that you have to call `CastToWritable` to be able to modify it.

There are two options here:

1. We make `CastToWritable` callable on an lvalue and have it do a `Retain()` internally: `    PacketBufferHandle CastToWritable() { return PacketBufferHandle::Retain(); }`
2. We add a `Retain` that returns `EncryptedPacketBufferHandle`: `EncryptedPacketBufferHandle Retain() { return EncryptedPacketBufferHandle(PacketBufferHandle::Retain()); }` and then we `CastToWritable` that thing that `Retain` returns.